agent/attachment.go

```diff
@@ -0,0 +1,103 @@
+package agent
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/identity"
+	"github.com/docker/swarmkit/log"
+	"golang.org/x/net/context"
+)
+
+// Listener allow to receive notifications about non executor attachment objects.
+type Listener interface {
```

Comment:
This package has lots of different types. Could you provide this with a more specific name or leverage the same listener model?